docs/1_how_it_works/1_1_how_it_works.md

```diff
@@ -0,0 +1,51 @@
+# How Finetuner Works
+
+## Contrastive Metric Learning 👓
+
+From the algorithmic perspective,
+**Finetuner** leverage contrastive metric learning approach to improve your model.
+How it works?
+
+### Step 1: Convert your model into an Embedding Model
+
+Finetuner interpret your model architecture,
+remove the default *head*, apply *pooling* and freeze layers not necessary to train.
+For example, if your pre-trained model is trained for cat-dog classification.
+Finetuner is going to remove the cat-dog classifier and turn your model into an *embedding model*.
+
+This *embedding model* do not make predictions and output a probability,
+but output a *feature vector* to represent your data.
+
+### Step 2: Triplets Construction and Training on-the-fly
+
+Finetuner "looks" into the label of your training data.
+Each `Document` and `Document`s share the same `finetuner_label` is considered as a *Positive*.
+In the meanwhile, each `Document` and `Documents` share a different `finetuner_label` is considered as a *Negative*.
+
+During model fine-tuning, Finetuner is creating *Triplets*  ``(anchor, positive, negative)`` on-the-fly.
+Finetuner then use the triplets to perform training,
+the objective is to pull `Document`s belongs to the same class together,
+while push the `Document`s belongs to the different class away from each other.
+
+
+
+## Cloud-based Fine-tuning ⛅
+
+From engineering perspective,
+we have hide all the complexity of machine learning algorithms,
+resources setup (such as GPU),
+all you need to do is to decide your backbone model and prepare your training data.
```

Comment:
"From an engineering perspective, we have hidden all the complexity of machine learning algorithms and resources configuration (such as GPU). All you need ... "